app/javascript/packages/document-capture/components/in-person-location-step.tsx

```diff
@@ -1,14 +1,190 @@
-import { PageHeading } from '@18f/identity-components';
-import { FormStepsButton } from '@18f/identity-form-steps';
+import { useState, useEffect, useCallback, useRef } from 'react';
 import { useI18n } from '@18f/identity-react-i18n';
+import {
+  PageHeading,
+  LocationCollectionItem,
+  LocationCollection,
+  SpinnerDots,
+} from '@18f/identity-components';
+
+interface PostOffice {
+  address: string;
+  city: string;
+  name: string;
+  phone: string;
+  saturday_hours: string;
+  state: string;
+  sunday_hours: string;
+  weekday_hours: string;
+  zip_code_4: string;
+  zip_code_5: string;
+}
+
+interface FormattedLocation {
+  formattedCityStateZip: string;
+  id: number;
+  name: string;
+  phone: string;
+  saturdayHours: string;
+  streetAddress: string;
+  sundayHours: string;
+  weekdayHours: string;
+}
+
+const locationUrl = '/verify/in_person/usps_locations';
+
+const getResponse = async () => {
+  const response = await fetch(locationUrl).then((res) =>
+    res.json().catch((error) => {
+      throw error;
+    }),
+  );
+  return response;
+};
+
+const formatLocation = (postOffices: PostOffice[]) => {
+  const formattedLocations = [] as FormattedLocation[];
+  postOffices.forEach((po: PostOffice, index) => {
+    const location = {
+      formattedCityStateZip: `${po.city}, ${po.state}, ${po.zip_code_5}-${po.zip_code_4}`,
+      id: index,
+      name: po.name,
+      phone: po.phone,
+      saturdayHours: po.saturday_hours,
+      streetAddress: po.address,
+      sundayHours: po.sunday_hours,
+      weekdayHours: po.weekday_hours,
+    } as FormattedLocation;
+    formattedLocations.push(location);
+  });
+  return formattedLocations;
+};
+
+const snakeCase = (value: string) =>
+  value
+    .split(/(?=[A-Z])/)
+    .join('_')
+    .toLowerCase();
+
+// snake case the keys of the location
+const prepToSend = (location: object) => {
+  const sendObject = {};
+  Object.keys(location).forEach((key) => {
+    sendObject[snakeCase(key)] = location[key];
+  });
+  return sendObject;
+};
 
 function InPersonLocationStep() {
   const { t } = useI18n();
+  const [locationData, setLocationData] = useState([] as FormattedLocation[]);
+  const [inProgress, setInProgress] = useState(false);
+  const [autoSubmit, setAutoSubmit] = useState(false);
+  const [isLoadingComplete, setIsLoadingComplete] = useState(false);
+
+  // ref allows us to avoid a memory leak
+  const mountedRef = useRef(false);
+
+  useEffect(() => {
+    mountedRef.current = true;
+    return () => {
+      mountedRef.current = false;
+    };
+  }, []);
+
+  // useCallBack here prevents unnecessary rerenders due to changing function identity
+  const handleLocationSelect = useCallback(
+    async (e: any, id: number) => {
```

Comment:
(Non-blocking) Can we avoid the `any` type here? I _think_ this would be the React `MouseEvent`.

e.g.

```ts
import type { MouseEvent } from 'react';
```

Reply:
good catch, in the locationCollectionItem props this type is React.FormEvent<HTMLInputElement> so I'll update it with that